edx_exams/apps/core/management/commands/test/test_bulk_add_course_staff.py

```diff
@@ -0,0 +1,77 @@
+""" Tests for bulk_add_course_staff management command """
+from tempfile import NamedTemporaryFile
+
+from django.core.management import call_command
+from django.test import TestCase
+
+from edx_exams.apps.core.models import CourseStaffRole
+from edx_exams.apps.core.test_utils.factories import UserFactory
+
+
+class TestBulkAddCourseStaff(TestCase):
+    """ Test bulk_add_course_staff management command """
+
+    def setUp(self):
+        super().setUp()
+        self.command = 'bulk_add_course_staff'
+        self.success_log_message = 'Bulk add course staff complete!'
+
+        # create existing user
+        self.user = UserFactory.create(
+            username='amy',
+            is_active=True,
+            is_staff=True,
+        )
+        self.user.email = 'amy@pond.com'
+        self.user.save()
+
+        self.course_id = 'course-v1:edx+test+f19'
+
+    def _write_test_csv(self, csv, lines):
+        """Write a test csv file with the lines provided"""
+        csv.write(b'username,email,role,course_id\n')
+        for line in lines:
+            csv.write(line.encode())
+        csv.seek(0)
+        return csv
+
+    def test_empty_csv(self):
+        lines = []
+        with NamedTemporaryFile() as csv:
+            csv = self._write_test_csv(csv, lines)
+            with self.assertNumQueries(1):
+                call_command(self.command, f'--csv_path={csv.name}')
+
+    def test_add_course_staff_with_existing_user(self):
+        lines = ['amy,amy@pond.com,staff,course-v1:edx+test+f19\n']
+        with NamedTemporaryFile() as csv:
+            csv = self._write_test_csv(csv, lines)
+            with self.assertNumQueries(2):
+                call_command(self.command, f'--csv_path={csv.name}')
+                assert CourseStaffRole.objects.filter(user=self.user.id).exists()
+
+    def test_add_course_staff_with_new_user(self):
+        lines = ['pam,pam@pond.com,staff,course-v1:edx+test+f20\n']
```

Comment:
I can try to refactor this to avoid repetition if that is better. Wasn't sure if it was more readable to have repeating lines variable for each test.